tests/conftest.py

```diff
@@ -350,43 +345,19 @@ def use_testdb(testdb, monkeypatch):
 
 
 @pytest.fixture()
-def s3_db_factory(tmpdir):
-    bucketname = str(uuid.uuid4())
-
-    def _s3_db_factory(keys, datasets=None, skip_metadata=False):
-        from terracotta import get_driver
-
-        with tempfile.TemporaryDirectory() as tmpdir:
-            dbfile = Path(tmpdir) / 'tc.sqlite'
-            driver = get_driver(dbfile)
-            driver.create(keys)
-
-            if datasets:
-                for keys, path in datasets.items():
-                    driver.insert(keys, path, skip_metadata=skip_metadata)
-
-            with open(dbfile, 'rb') as f:
-                db_bytes = f.read()
-
-        conn = boto3.resource('s3')
-        conn.create_bucket(Bucket=bucketname)
-
-        s3 = boto3.client('s3')
-        s3.put_object(Bucket=bucketname, Key='tc.sqlite', Body=db_bytes)
-
-        return f's3://{bucketname}/tc.sqlite'
+def use_non_writable_testdb(testdb, monkeypatch, raster_file):
+    import terracotta
+    terracotta.update_settings(DRIVER_PATH=str(testdb))
 
-    return _s3_db_factory
+    driver = terracotta.get_driver(testdb)
+    with driver.connect():
+        driver.insert(('first', 'second', 'third'), str(raster_file), skip_metadata=True)
 
+    driver.meta_store._WRITABLE = False
+    yield
+    driver.meta_store._WRITABLE = True
```

Comment:
You already give monkeypatch as a fixture, so why not use it? :smile: 

```suggestion
    with monkeypatch.context() as m:
        m.setattr(driver.meta_store, "_WRITABLE", False)
        yield
```

(no need to set it back)

Reply:
Huh, TIL! :clap: